### DISCUSSION TITLE
[Bitbucket] Config flag `bbUseDevelopmentBranch` doesn't work (main branch used instead)

### DISCUSSION THREAD

**Discussion by matemijolovic** [renovatebot/renovate#26805] [Request Help]
### How are you running Renovate?

Self-hosted Renovate

### If you're self-hosting Renovate, tell us what version of Renovate you run.

37.141.1

### If you're self-hosting Renovate, select which platform you are using.

Bitbucket Cloud (bitbucket.org)

### Was this something which used to work for you, and then stopped?

I have not seen this working

### Describe the problem

Hello, 

I'm trying to set up Renovate to work with Bitbucket Cloud repositories that have a `DEVELOPMENT` branch set up (via Bitbucket UI). I've tried adding the flag `bbUseDevelopmentBranch` to the config but it didn't seem to have any effect.

Minimal reproduction:
1. Create a Bitbucket Cloud repository with separate `main` and `develop` branches (annotated as `MAIN` and `DEVELOPMENT`in the repository settings, respectively) - or, alternatively, fork any repo with a separate development branch
3. Add `renovate.json` config to the `develop` branch
4. Connect Renovate to the given repository and enable config flag `bbUseDevelopmentBranch` (EDIT: to clarify, the flag is added to the global config)

I've tracked down the issue:
1. The BB endpoint `/2.0/repositories/some/repo/branching-model` returns the correct data and the config entry `defaultBranch` is correctly set to a development branch
5. The bug seems to live here: `syncGit` command doesn't respect the `defaultBranch` setting when cloning the PR ([source](https://github.com/renovatebot/renovate/blob/main/lib/util/git/index.ts#L429))  

The fix seems simple: just add additional opt `-b <default_branch>` and fix how `config.currentBranch` is set afterwards.

Pull request: https://github.com/renovatebot/renovate/pull/26810

### Relevant debug logs

<details><summary>Details</summary>
<p>

I've masked the sensitive data
```
DEBUG: Using localDir: /var/folders/2s/96lg10vj4_v8rt7ntb5dkp9w0000gn/T/renovate/repos/bitbucket/org/repo (repository=org/repo)
DEBUG: PackageFiles.clear() - Package files deleted (repository=org/repo)
DEBUG: initRepo("org/repo") (repository=org/repo)
DEBUG: hostRules: applying Basic authentication for api.bitbucket.org (repository=org/repo)
DEBUG: org/repo owner = org (repository=org/repo)
DEBUG: Resetting npmrc (repository=org/repo)
DEBUG: Resetting npmrc (repository=org/repo)
DEBUG: checkOnboarding() (repository=org/repo)
DEBUG: isOnboarded() (repository=org/repo)
DEBUG: findPr(renovate/configure, Configure Renovate, !open) (repository=org/repo)
DEBUG: getPrList() (repository=org/repo)
DEBUG: Syncing PR list (repository=org/repo)
DEBUG: findPr(renovate/configure, chore: Configure Renovate, !open) (repository=org/repo)
DEBUG: getPrList() (repository=org/repo)
DEBUG: Found PR #1319 (repository=org/repo)
DEBUG: findFile(renovate.json) (repository=org/repo)
DEBUG: Initializing git repository into /var/folders/2s/96lg10vj4_v8rt7ntb5dkp9w0000gn/T/renovate/repos/bitbucket/org/repo (repository=org/repo)
DEBUG: Performing blobless clone (repository=org/repo)
DEBUG: git clone completed (repository=org/repo)
       "durationMs": 38939
DEBUG: latest repository commit (repository=org/repo)
       "latestCommit": ***
DEBUG: findFile(renovate.json5) (repository=org/repo)
DEBUG: findFile(.github/renovate.json) (repository=org/repo)
DEBUG: findFile(.github/renovate.json5) (repository=org/repo)
DEBUG: findFile(.gitlab/renovate.json) (repository=org/repo)
DEBUG: findFile(.gitlab/renovate.json5) (repository=org/repo)
DEBUG: findFile(.renovaterc) (repository=org/repo)
DEBUG: findFile(.renovaterc.json) (repository=org/repo)
DEBUG: findFile(.renovaterc.json5) (repository=org/repo)
DEBUG: config file not found (repository=org/repo)
DEBUG: Found closed onboarding PR (repository=org/repo)
DEBUG: Repo is not onboarded and no merged PRs exist (repository=org/repo)
DEBUG: Found 3 comments (repository=org/repo)
DEBUG: Ensuring comment "Renovate is disabled" in #1319 (repository=org/repo)
DEBUG: Comment is already update-to-date (repository=org/repo)
 INFO: Repository is disabled - skipping (repository=org/repo)
DEBUG: Repository result: disabled-closed-onboarding, status: disabled, enabled: false, onboarded: undefined (repository=org/repo)
```

</p>
</details> 

### Have you created a minimal reproduction repository?

I have linked to a minimal reproduction in the description above

**Comment by viceice** [comment]
the renovate.json needs to be on default branch or set that option on global bot config.

**Comment by matemijolovic** [comment]
I put the option in the global config and I can confirm it's enabled in the debugger.
EDIT: also, default branch is `develop` in my case and it contains a `renovate.json`

**Comment by matemijolovic** [comment]
Hi, after digging a bit deeper today I found out this is also related to the `baseBranches` global config setting. I've included the test that triggers the undesired behaviour in the PR.

**Comment by matemijolovic** [comment]
Resolved in https://github.com/renovatebot/renovate/pull/26810
